### PR TITLE
feat(auth): add option to enforce second factor authentication

### PR DESCRIPTION
**Pull request by nijel**
## Proposed changes

Built on top of https://github.com/WeblateOrg/weblate/pull/12173
   
    
- projects can define this and it will block users from editing
- teams can define this and roles from the teams will not be applied unless user is 2fa verified
- middleware warns user about not meeting this requirement

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added documentation to describe my feature.
- [ ] I have squashed my commits into logic units.
- [ ] I have described the changes in the commit messages.

### CODE REVIEW

**Comment by orangesunny** [comment]
@nijel, I would wait with a review until 12173 is merged, as it touches the same files and I did some corrections there. I would rather not get lost or create conflicts.

**Comment by nijel** [comment]
Ready for review, #12173 is merged and this one rebased on top of that.

**Review by orangesunny — APPROVED** [review]
Good to go; can’t wait to use it!

File: docs/admin/projects.rst [hunk 1]
```diff
@@ -200,6 +200,16 @@ Configure per project access control, see :ref:`acl` for more details.
 
 The default value can be changed by :setting:`DEFAULT_ACCESS_CONTROL`.
 
+.. _project-enforced_2fa:
+
+Enforced two factor authentication
```
Comment (nijel) [inline docs/admin/projects.rst:205]:
```suggestion
Enforced two-factor authentication
```

**Pull request merged**